ibis/backends/duckdb/tests/test_geospatial.py

```diff
@@ -0,0 +1,197 @@
+        param("n_points", {}, id="n_points"),
+    ],
+)
+def test_geospatial_unary_snapshot(operation, keywords, snapshot):
+    t = ibis.table([("geom", "geometry")], name="t")
+    expr = getattr(t.geom, operation)(**keywords).name("tmp")
+    snapshot.assert_match(ibis.to_sql(expr), "out.sql")
+
+
+def test_geospatial_dwithin(snapshot):
+    t = ibis.table([("geom", "geometry")], name="t")
+    expr = t.geom.d_within(t.geom, 3.0).name("tmp")
+
+    snapshot.assert_match(ibis.to_sql(expr), "out.sql")
+
+
+# geospatial unary functions that return a non-geometry series
+# we can test using pd.testing (tm)
+@pytest.mark.parametrize(
+    ("op", "keywords", "gp_op"),
+    [
+        param("area", {}, "area", id="area"),
+        param("is_valid", {}, "is_valid", id="is_valid"),
+        param(
+            "geometry_type",
+            {},
+            "geom_type",
+            id="geometry_type",
+            marks=pytest.mark.xfail(raises=pa.lib.ArrowTypeError),
+        ),
+    ],
+)
+def test_geospatial_unary_tm(op, keywords, gp_op, zones, zones_gdf):
+    expr = getattr(zones.geom, op)(**keywords).name("tmp")
+    gp_expr = getattr(zones_gdf.geometry, gp_op)
+
+    tm.assert_series_equal(expr.to_pandas(), gp_expr, check_names=False)
+
+
+@pytest.mark.parametrize(
+    ("op", "keywords", "gp_op"),
+    [
+        param("x", {}, "x", id="x_coord"),
+        param("y", {}, "y", id="y_coord"),
+    ],
+)
+def test_geospatial_xy(op, keywords, gp_op, zones, zones_gdf):
+    cen = zones.geom.centroid().name("centroid")
+    gp_cen = zones_gdf.geometry.centroid
+
+    expr = getattr(cen, op)(**keywords).name("tmp")
+    gp_expr = getattr(gp_cen, gp_op)
+
+    tm.assert_series_equal(expr.to_pandas(), gp_expr, check_names=False)
+
+
+def test_geospatial_length(lines, lines_gdf):
+    # note: ST_LENGTH returns 0 for the case of polygon
+    # or multi polygon while pandas geopandas returns the perimeter.
+    length = lines.geom.length().name("length")
+    gp_length = lines_gdf.geometry.length
+
+    tm.assert_series_equal(length.to_pandas(), gp_length, check_names=False)
+
+
+# geospatial binary functions that return a non-geometry series
+# we can test using pd.testing (tm)
+@pytest.mark.parametrize(
+    ("op", "keywords", "gp_op", "gp_keywords"),
+    [
+        param("contains", {}, "contains", {}, id="contains"),
+        param("geo_equals", {}, "geom_equals", {}, id="geo_eqs"),
+        param("covers", {}, "covers", {}, id="covers"),
+        param("covered_by", {}, "covered_by", {}, id="covered_by"),
+        param("crosses", {}, "crosses", {}, id="crosses"),
+        param("disjoint", {}, "disjoint", {}, id="disjoint"),
+        param("distance", {}, "distance", {}, id="distance"),
+        param("intersects", {}, "intersects", {}, id="intersects"),
+        param("overlaps", {}, "overlaps", {}, id="overlaps"),
+        param("touches", {}, "touches", {}, id="touches"),
+        param("within", {}, "within", {}, id="within"),
```

Comment:
Do we need the future-proofing of the empty kwargs dict?  Or were there options that were being used that are gone now?

Reply:
Good catch. I think we can remove them all, if in the future something gets added as a kwarg we can test for it and add it.